### PR TITLE
read image import CAs from mounted configmap

### PR DESCRIPTION
**Pull request by bparees**
configmap is mounted by https://github.com/openshift/cluster-openshift-apiserver-operator/pull/89

### CODE REVIEW

**Comment by bparees** [comment]
/hold

@adambkaplan I still need to test this locally, but you can go ahead and review.

**Comment by bparees** [comment]
/hold cancel

tested this locally, was able to confirm image import fails if i overwrote all the /etc/pki certs and then succeeds when i mounted a configmap containing a valid cert.pem into `/var/run/configmaps/image-import-ca`

File: pkg/image/apiserver/apiserver.go [hunk 1]
```diff
@@ -144,10 +149,29 @@ func (c *completedConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
 		tlsConfig.RootCAs = x509.NewCertPool()
 	}
 
-	if len(c.ExtraConfig.AdditionalTrustedCA) != 0 {
-		if ok := tlsConfig.RootCAs.AppendCertsFromPEM(c.ExtraConfig.AdditionalTrustedCA); !ok {
-			return nil, fmt.Errorf("No valid certificates read from %v", c.ExtraConfig.AdditionalTrustedCA)
+	err = filepath.Walk("/var/run/configmaps/image-import-ca", func(path string, info os.FileInfo, err error) error {
```
Comment (adambkaplan) [inline pkg/image/apiserver/apiserver.go]: Can this error if the path does not exist? Noticed that the mount at this path is optional.

Comment (bparees) [reply pkg/image/apiserver/apiserver.go]: i'm not sure, but the dir will always exist even if the configmap does not, it'll just be empty (i just checked this).

and worst case an error from filepath.Walk is not being treated as fatal, we just log it.

**Pull request merged**